crates/ruff_python_semantic/src/analyze/visibility.rs

```diff
@@ -12,6 +13,41 @@ pub enum Visibility {
     Private,
 }
 
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub enum AbstractDecoratorKind {
+    /// `abc.abstractmethod`
+    AbstractMethod,
+    /// `abc.abstractclassmethod`
+    AbstractClassMethod,
+    /// `abc.abstractstaticmethod`
+    AbstractStaticMethod,
+    /// `abc.abstractproperty`
+    AbstractProperty,
+}
+
+impl AbstractDecoratorKind {
+    fn from(name: &str) -> Option<Self> {
+        match name {
+            "abstractmethod" => Some(Self::AbstractMethod),
+            "abstractclassmethod" => Some(Self::AbstractClassMethod),
+            "abstractstaticmethod" => Some(Self::AbstractStaticMethod),
+            "abstractproperty" => Some(Self::AbstractProperty),
+            _ => None,
+        }
+    }
```

Comment:
Nit: I suggest implementing `from(name: &QualifiedName)` because the method returns an `Option` anyway and the decorator is only "unique" if it's also the right module